default_speeds.json

```diff
@@ -1,6281 +1,6050 @@
 {
     "AE": {
         "(default)": {
-            "(default)": [
-                "100"
-            ],
-            "bus": [
-                "80"
-            ],
-            "hgv": [
-                "80"
-            ]
-        },
-        "motorway": {
-            "(default)": [
-                "120"
-            ],
-            "bus": [
-                "80"
-            ],
-            "hgv": [
-                "80"
-            ]
-        },
-        "service road": {
-            "(default)": [
-                "25"
-            ]
+            "(default)": {
+                "maxspeed": "100"
+            },
+            "bus": {
+                "maxspeed": "80"
+            },
+            "hgv": {
+                "maxspeed": "80"
+            }
+        },
+        "highway=service": {
+            "(default)": {
+                "maxspeed": "25"
+            }
+        },
+        "highway~^motorway|motorway_link$": {
+            "(default)": {
+                "maxspeed": "120"
+            },
+            "bus": {
+                "maxspeed": "80"
+            },
+            "hgv": {
+                "maxspeed": "80"
+            }
         },
         "urban dual carriageway": {
-            "(default)": [
-                "60"
-            ]
+            "(default)": {
+                "maxspeed": "60"
+            }
         },
         "urban single carriageway": {
```

Comment:
First, maybe I am mixing something up, but should the road type name at this point not already be replaced with the tags from the second table or dropped if it cannot be mapped? Or is it just left as-is if it cannot be mapped?

(I know, we talked about maybe not mapping the road types to tags already in this file but instead in a second one to make it possible to inject a fuzzy mapping, I am just commenting on the state of the code as it is now. )

Second, with "urban single carriageway" road type specifically and some others, the mapping on the wiki page includes `???`. I put it this way into the wiki page rather than just leaving it for two reasons:
1. so that one can immediately see on the page that some explicit key is missing
2. because part of the keys it should map to are already clear, i.e. in this case only what constitutes "urban" is not clear, but the rest is: `??? and (oneway!~^yes|-1$ or dual_carriageway=no)`.

So the parser should probably either treat this as "can not map", or I change the wiki page to leave out the keys that can (partially) not be mapped out completely from the second table.